decompiler/config/jak2/all-types.gc

```diff
@@ -25472,37 +25618,35 @@
    (backup-nav-radius       float                    :offset-assert 44)
    (cam-side-shift          float                    :offset-assert 48)
    (enable-cam-side-shift   basic                    :offset-assert 52)
-   (gun?                    basic                    :offset-assert 56)
-   (controls                vehicle-controls :inline :offset-assert 60)
-   (accel-array             vector           :inline :offset-assert 80)
-   (local-accel             vector           :inline :offset-assert 208)
-   (pilot-trans             vector           :inline :offset-assert 224)
-   (pilot-quat              vector           :inline :offset-assert 240)
-   (pilot-scale             vector           :inline :offset-assert 256)
-   (pilot-time              time-frame                   :offset-assert 272)
-   (as-daxter?              basic                    :offset-assert 280)
-   (art-group-backup        basic                    :offset-assert 284)
+   (gun?                    symbol                   :offset-assert 56)
+   (controls                vehicle-controls         :offset 60)
```

Comment:
This is one is probably actually inline. You can tell because there's a large gap between this field and the next (20 bytes) and the `method 3 pilot-info` does
```
daddiu a1, fp, L16        ;; [ 91] (set! a1 L16) "~1Tcontrols: #<vehicle-controls @ #x~X>~%"
daddiu a2, gp, 56         ;; [ 92] (set! a2 (+ gp 56))
```

It uses a `daddiu` to add an offset of 56 bytes to the object, so it's getting the address of something stored inline.  If it wasn't inline, it would be a `lwu a2, 56(gp)`.

Reply:
The decompiler will complain that `vehicle-controls` should be placed at 64-bytes if you make it inline.  By default, inline structures must be 16-byte aligned, so it doesn't work at an offset of 60. We have to manually add a `:pack-me` flag to tell the compiler that it's okay to have this be misaligned.  Then the minimum alignment of `vehicle-controls` will become 4-bytes (the size of a float)

If you add `:pack-me` to `vehicle-controls`, it should be fine. (Note that you have to add it to `vehicle-controls`, not `pilot-info`).